credentials/tls/certprovider/pemfile/watcher.go

```diff
@@ -78,6 +79,10 @@ func (o Options) canonical() []byte {
 }
 
 func (o Options) validate() error {
+	// Guard against SPIFFE bundle map usage
+	if os.Getenv(spiffeEnabledEnvVar) != "true" {
```

Comment:
Rather than overwrite any SPIFFE bundle map, should we prevent populating it in the first place if the env variable is not true? Similarly below.

Reply:
It's not really easy to prevent populating it - these options are coming by settings the option on a struct literal. However, this validate call happens on the `Options` before a provider is made, thereby preventing populating it

Do you have another idea of how this should be done?

Reply:
Ack understood, that should be fine then. Thanks!